File: frontend/src/api/projectService.js
import api from './axios';

export const projectAPI = {
  submit: (data) => api.post('/projects', data),
  update: (id, data) => api.put(`/projects/${id}`, data),
  getById: (id) => api.get(`/projects/${id}`),
  getByHackathon: (hackathonId) => api.get(`/projects/hackathon/${hackathonId}`),
  getByTeam: (teamId) => api.get(`/projects/team/${teamId}`),
  delete: (id) => api.delete(`/projects/${id}`),
};

File: backend/src/main/java/com/hackathon/service/ProjectService.java
package com.hackathon.service;

import com.hackathon.dto.request.ProjectRequest;
import com.hackathon.model.Project;

import java.util.List;

public interface ProjectService {

    Project submit(ProjectRequest request, String userId);

    Project update(String id, ProjectRequest request, String userId);

    Project getById(String id);

    List<Project> getByHackathonId(String hackathonId);

    Project getByTeamId(String teamId);

    void delete(String id, String userId);
}
